closed/src/java.base/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -136,9 +136,13 @@ private void init(Key key)
                 this.initializeJavaImplementation(key);
                 return;
             }
-            this.privateKeyOps = opsOpt.get();
 
-            ECParameterSpec params = this.privateKey.getParams();
+            if (NativeECUtil.isBrainpoolP512r1(ecPrivateKey.getParams())) {
+                this.privateKeyOps = null;
+            } else {
+                this.privateKeyOps = ops;
+            }
```

Comment:
Part of the point of having `ops` was to be able to use it directly here:
```java
            this.privateKeyOps = ops;
```
and avoid calling `isBrainpoolP512r1()` a second time. We expect `ops` to be null (for brainpoolP512r1): is it a problem if it's not?

Reply:
I think this can work as you describe just setting `this.privateKeyOps = ops;`. I do expect `ops` to be null for brainpool and be set correctly otherwise.